corehq/apps/reports/standard/forms/reports.py

```diff
@@ -59,6 +60,16 @@ def submitfilter(self):
             self._submitfilter = SubmissionTypeFilter.get_filter_toggle(self.request)
         return self._submitfilter
 
+    @property
+    @memoized
+    def selected_form_data(self):
+        forms = list(FormsByApplicationFilter.get_value(self.request, self.domain).values())
+        if len(forms) == 1 and forms[0]['xmlns']:
+            return forms[0]
+        non_fuzzy_forms = [form for form in forms if not form['is_fuzzy']]
+        if len(non_fuzzy_forms) == 1:
+            return non_fuzzy_forms[0]
```

Comment:
@zandre-eng What exactly is a fuzzy/non-fuzzy form?

Reply:
A fuzzy form in this context are filter results where the XMLNS cannot be matched to exactly one application. For more context, you can have a look at [this](https://github.com/dimagi/commcare-hq/blob/2d72e13a5d8d1ce01da74c12fae652cb3bc7b3c5/corehq/apps/reports/filters/forms.py#L86) docstring for the `FormsByApplicationFilter` filter.